appshell/node-core/Server.js

```diff
@@ -199,7 +199,20 @@ maxerr: 50, node: true */
                         stop();
                     });
                     
-                    wsServer = new WebSocket.Server({server: httpServer});
+                    wsServer = new WebSocket.Server({
+                        server: httpServer,
+                        verifyClient : function (info, callback) {
+                            // Accept connections originated from local system only
+                            // Also do a loose check on user-agent to accept connection only from Brackets CEF shell
+                            if (info.origin === "file://" && info.req.headers["user-agent"].indexOf(" Brackets") !== -1) {
```

Comment:
Yes. The "user-agent" field looks like this in case we raise a request from Brackets shell 
`Mozilla/5.0 (Windows NT 6.2; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Brackets/1.6.0.0 Chrome/39.0.2171.36 Safari/537.36`